src/spatial/DirectCRonDirect.jl

```diff
@@ -78,37 +78,27 @@ function aggregate(aggregator::DirectCRonDirect, starting_state, p, t, end_time,
     hop_rates = HopRates(hopping_constants)
     site_rates = zeros(typeof(end_time), num_sites(spatial_system))
 
-    DirectCRonDirectJumpAggregation(next_jump, next_jump_time, end_time, rx_rates, hop_rates, site_rates, save_positions, rng, spatial_system; num_specs = num_species, kwargs...)
+    DirectCRDirectJumpAggregation(next_jump, next_jump_time, end_time, rx_rates, hop_rates, site_rates, save_positions, rng, spatial_system; num_specs = num_species, kwargs...)
 end
 
 # set up a new simulation and calculate the first jump / jump time
-function initialize!(p::DirectCRonDirectJumpAggregation, integrator, u, params, t)
+function initialize!(p::DirectCRDirectJumpAggregation, integrator, u, params, t)
     fill_rates_and_get_times!(p, u, t)
     generate_jumps!(p, integrator, params, u, t)
     nothing
 end
 
 # calculate the next jump / jump time
-function generate_jumps!(p::DirectCRonDirectJumpAggregation, integrator, params, u, t)
-    @unpack rx_rates, hop_rates, rng, rt = p
-
-    p.next_jump_time  = t + randexp(rng) / rt.gsum
-    
-    if p.next_jump_time < p.end_time
-        site = sample(p.rt, p.site_rates, p.rng)
-        if rand(rng)*(total_site_rate(rx_rates, hop_rates, site)) < total_site_rx_rate(rx_rates, site)
-            rx = sample_rx_at_site(rx_rates, site, rng)
-            p.next_jump = SpatialJump(site, rx+p.numspecies, site)
-        else
-            species_to_diffuse, target_site = sample_hop_at_site(hop_rates, site, rng, p.spatial_system)
-            p.next_jump = SpatialJump(site, species_to_diffuse, target_site)
-        end
-    end    
+function generate_jumps!(p::DirectCRDirectJumpAggregation, integrator, params, u, t)
+    p.next_jump_time  = t + randexp(rng) / p.rt.gsum
+    p.next_jump_time >= p.end_time && return nothing
+    site = sample(p.rt, p.site_rates, p.rng)
+    p.next_jump = sample_jump_direct(p, site)
```

Comment:
Maybe call this something like `sample_jump_at_site_direct`?

Reply:
Actually the `direct` isn’t needed; other appraoches could use the same function and just dispatch (like RSSA).